.github/workflows/check.yml

```diff
@@ -26,15 +26,16 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-node@v3
+        id: setup-node
         with:
-          node-version: '16'
+          node-version-file: 'package.json'
```

Comment:
This uses `package.json`'s `engines` to determine which node version to use.

This way we'll have less places to touch next time we upgrade.

Reply:
This would mean Node with `^20` version, which will change over time. There may be minor changes, maybe bugs. While in the development environment we've pinned `20.9.0` in  `.nvmrc`.

I'm leaning towards using `node-version-file: '.nvmrc'` to keep the versions consistent across the environments. An alternative would be Volta which we don't use.

Reply:
Good points, but the downside of `.nvmrc` is it's not really enforced. 
What do you think of pinning node version in the `engines` field then?